lib/SGN/Controller/AJAX/Search/Cross.pm

```diff
@@ -69,21 +69,62 @@ sub search : Path('/ajax/search/cross_info') Args(0) {
 
 
 
-    my $q = "SELECT female_parent.stock_id, male_parent.stock_id, cross_entry.stock_id, female_parent.uniquename,male_parent.uniquename,cross_entry.uniquename FROM stock as female_parent
-    INNER JOIN stock_relationship AS stock_relationship1 ON (female_parent.stock_id=stock_relationship1.subject_id)
+    my $q = "SELECT female_parent.stock_id, male_parent.stock_id, cross_entry.stock_id, female_parent.uniquename, male_parent.uniquename, cross_entry.uniquename, stock_relationship1.value
+    FROM stock as female_parent INNER JOIN stock_relationship AS stock_relationship1 ON (female_parent.stock_id=stock_relationship1.subject_id)
+    AND stock_relationship1.type_id='$female_parent_typeid' INNER JOIN stock_relationship AS stock_relationship2
+    ON (stock_relationship1.object_id=stock_relationship2.object_id) INNER JOIN stock AS male_parent
+    ON (male_parent.stock_id=stock_relationship2.subject_id) AND stock_relationship2.type_id='$male_parent_typeid'
+    INNER JOIN stock AS cross_entry ON (cross_entry.stock_id=stock_relationship2.object_id) AND cross_entry.type_id='$cross_typeid'
+    WHERE female_parent.uniquename = '$female_parent' AND male_parent.uniquename = '$male_parent' ORDER BY stock_relationship1.value, male_parent.uniquename ASC";
+
+    my $h = $dbh->prepare($q);
+    $h->execute();
```

Comment:
for parameters coming from outside of a function, especially those coming from $c->req->param, should be bound to your query using placeholders. you should fix this for all $h->execute() in this file.